### PR TITLE
Add copy button to dashboard and visualisation

### PR DESCRIPTION
**Pull request by h-kanazawa**
#### Which issue does this close?
Closes https://learningpool.atlassian.net/browse/LL-26

### CODE REVIEW

File: ui/src/containers/Dashboard/index.js [hunk 1]
```diff
@@ -1,29 +1,39 @@
...
 
 class Dashboard extends Component {
   static propTypes = {
     model: PropTypes.instanceOf(Map),
-    updateModel: PropTypes.func
+    navigateTo: PropTypes.func,
+    updateModel: PropTypes.func,
+    saveModel: PropTypes.func,
+    setMetadata: PropTypes.func,
+    getMetadata: PropTypes.func,
```
Comment (ryansmith94) [inline ui/src/containers/Dashboard/index.js]: I can't see `getMetadata`, `setMetadata`, `saveModel`, and `navigateTo` being used, we might want to remove these?

File: ui/src/containers/Visualise/CopyButton.js [hunk 2]
```diff
@@ -0,0 +1,73 @@
+import React, { useState, useCallback } from 'react';
```
Comment (ryansmith94) [inline ui/src/containers/Visualise/CopyButton.js]: Thank you for using hooks in this new component 👍

File: ui/src/containers/Visualise/CopyButton.js [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+import { compose, withHandlers, mapProps } from 'recompose';
+import { withModel } from 'ui/utils/hocs';
+import ConfirmModal from 'ui/components/Modal/ConfirmModal';
+import { loggedInUserId } from 'ui/redux/modules/auth';
+import { COPY_VISUALISATION } from 'ui/redux/modules/visualisation/copyVisualisation';
+
+/**
+ * @param {function} _.copyVisualisation
+ */
+const CopyIconButton = ({ copyVisualisation }) => {
```
Comment (ryansmith94) [inline ui/src/containers/Visualise/CopyButton.js]: Do you think we should extract the copy button out in the dashboard component like we've done here for the visualisations?

File: ui/src/containers/Dashboard/index.js (hunk 1 above)
Comment (h-kanazawa) [reply ui/src/containers/Dashboard/index.js]: sorry. yes.

File: ui/src/containers/Visualise/CopyButton.js (hunk 3 above)
Comment (h-kanazawa) [reply ui/src/containers/Visualise/CopyButton.js]: ok

File: ui/src/containers/Dashboard/index.js (hunk 1 above)
Comment (h-kanazawa) [reply ui/src/containers/Dashboard/index.js]: - `setMetadata` and `getMetadata` is used in toggleSharing (line 137)
- `saveModel` is used createPopulatedWidget (line 117)
- `navigateTo` is passed to DeleteButton (line 187)

Comment (ryansmith94) [reply ui/src/containers/Dashboard/index.js]: Okay cool thank you

**Pull request merged**